specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/networkWatcher.json

```diff
@@ -3192,6 +3192,10 @@
           "description": "ID of the storage account which is used to store the flow log.",
           "type": "string"
         },
+        "enabledFilteringCriteria": {
```

Comment:
Why are you calling it 'enabledFilteringCriteria'?

One way I'm puzzled by it - If you had any filtering criteria, why wouldn't you want to enable them?

Reply:
Also, are there any special reasons you went with the mini language that you, or other processors of, will have to parse / emit? 

While it might seem pretty cool or simple that you can just write a nice powerful expression manually, (like the power of regex) it may not be as simple or pleasant to use in other contexts, like authoring ARM templates (where they might have to use a lot of 'concat'), or parsing your GET responses to fully analyze the current filtering state of the resource.

Reply:
We are a logging service. By default, we capture all the logs of customer's network traffic. This is the scenario which you can call as filtering 'disabled'. In this case, customer will not pass anything to this field value.
--
The reason we are using this expression is due to the nature of logs. We capture 5 tuple information (srcIP, destIP, srcPort, DestPort and Protocol). So, filtering criteria could be a compound condition of these field. Mathematical expression consisting of logical and relational operators best suits this. This is also the way filtering is provided in our competitive services in google cloud and aws. This has been discussed internally within team with Eng Leads and PMs.

Reply:
So its being called 'enabled' because its absence implies logging is 'disabled'? Or because the things matching the filer are 'enabled' for logging (logged), instead of 'disabled'? How can we make this clearer? Updating the description to explain whether you're filtering in, or filtering out, and what happens by default seems like the least you could do. 

I still wonder if its worth trying to come up with a better name, but I'm not sure I have any yet.

Reply:
Sure i'll update the description of the field and make it more verbose.
Regarding field name, we have already discussed this with the PMs.

Reply:
Updated